bundle-workflow/src/s3_utility.py

```diff
@@ -5,31 +5,35 @@
 # compatible open source license.
 
 import os
+from pathlib import Path
+from urllib.parse import urlparse
 
 import boto3
-
 from botocore.exceptions import ClientError
-from pathlib import Path
-from urllib.parse import urlparse
 
 
 def file_download_helper(bucket, key, path):
     try:
         bucket.download_file(key, path)
-    except ClientError as e: # TODO: handle the right exception
-        if e.response['Error']['Code'] == "404":
+    except ClientError as e:  # TODO: handle the right exception
+        if e.response["Error"]["Code"] == "404":
             print("The object does not exist.")
         else:
             raise
 
 
 class S3ReadWrite:
```

Comment:
Rename class. Is this a bucket?

Reply:
It's a utility class that provides methods to read/write files/folders to/from s3. Sure, any suggestions? I could think of S3Wrapper, what else?

Reply:
It represents an S3 bucket, but reads and writes in a certain peculiar way (assumes role). I would put this in a new namespace called `aws.s3` and call it a `Bucket`.

Reply:
It does not represent an S3 bucket. It is like a wrapper over s3 boto apis - that wraps GetObject, PutObject apis to provide more specific abstractions for the purpose of test-infrastructure - download_folder, download_file, upload_file. You can provide uri related to any S3 bucket

Reply:
Then it should be all static methods and called, dunno, `boto4`? I think abstracting this away in OO is cleaner.